server/src/main/java/org/opensearch/gateway/remote/model/RemoteDiscoveryNodes.java

```diff
@@ -0,0 +1,100 @@
+import java.io.InputStream;
+import java.util.List;
+
+import static org.opensearch.core.common.bytes.BytesReference.toBytes;
+import static org.opensearch.gateway.remote.RemoteClusterStateAttributesManager.CLUSTER_STATE_ATTRIBUTES_CURRENT_CODEC_VERSION;
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.CLUSTER_STATE_EPHEMERAL_PATH_TOKEN;
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.DELIMITER;
+
+/**
+ * Wrapper class for uploading/downloading {@link DiscoveryNodes} to/from remote blob store
+ */
+public class RemoteDiscoveryNodes extends AbstractRemoteWritableBlobEntity<DiscoveryNodes> {
+
+    public static final String DISCOVERY_NODES = "nodes";
+
+    private DiscoveryNodes discoveryNodes;
+    private long stateVersion;
+
+    public RemoteDiscoveryNodes(
+        final DiscoveryNodes discoveryNodes,
+        final long stateVersion,
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        this.discoveryNodes = discoveryNodes;
+        this.stateVersion = stateVersion;
+    }
+
+    public RemoteDiscoveryNodes(
+        final String blobName,
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        this.blobName = blobName;
+    }
+
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        return new BlobPathParameters(List.of(CLUSTER_STATE_EPHEMERAL_PATH_TOKEN), DISCOVERY_NODES);
+    }
+
+    @Override
+    public String generateBlobFileName() {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/ephemeral/<componentPrefix>__<inverted_state_version>__<inverted__timestamp>__<codec_version>
+        String blobFileName = String.join(
+            DELIMITER,
+            getBlobPathParameters().getFilePrefix(),
+            RemoteStoreUtils.invertLong(stateVersion),
+            RemoteStoreUtils.invertLong(System.currentTimeMillis()),
+            String.valueOf(CLUSTER_STATE_ATTRIBUTES_CURRENT_CODEC_VERSION)
+        );
+        this.blobFileName = blobFileName;
+        return blobFileName;
+    }
+
+    @Override
+    public UploadedMetadata getUploadedMetadata() {
+        assert blobName != null;
+        return new UploadedMetadataAttribute(DISCOVERY_NODES, blobName);
+    }
+
+    @Override
+    public InputStream serialize() throws IOException {
+        BytesStreamOutput outputStream = new BytesStreamOutput();
+        discoveryNodes.writeTo(outputStream);
+        return outputStream.bytes().streamInput();
+    }
+
+    @Override
+    public DiscoveryNodes deserialize(final InputStream inputStream) throws IOException {
+        return DiscoveryNodes.readFrom(new BytesStreamInput(toBytes(Streams.readFully(inputStream))), null);
```

Comment:
The second param localNodeId would be needed to set in discovery nodes object unless we are setting it somewhere later.

Reply:
Yeah, as we can't pass localNodeId to deserialize method. We need to set it after getting the DiscoveryNodes object.